### ISSUE TITLE
Normalization/unnormalization is silently skipped after Pipeline migration for multi-dataset models

### ISSUE THREAD

**Issue by ceyxasm** [huggingface/lerobot#4415]
After the Pipeline migration (https://huggingface.co/lerobot/smolvla_base/discussions/11), normalization and unnormalization are  silently skipped during inference for models trained on multiple datasets (e.g., lerobot/smolvla_base).

### Root cause
The migration extracts normalization stats from the model's built-in normalize_inputs / unnormalize_outputs layers into external  PolicyProcessorPipeline safetensors files. For multi-dataset models, the stats are saved with dataset-prefixed keys:

* so100.buffer.action.mean
* so100.buffer.action.std
* so100-blue.buffer.action.mean
...

After load_state_dict, these become _tensor_stats keys like "so100.buffer.action".
However, _normalize_action() calls _apply_transform() with key=ACTION ("action"), and _apply_transform checks:
```
if norm_mode == NormalizationMode.IDENTITY or key not in self._tensor_stats:
    return tensor  # <-- silently returns unchanged
```
"action" is not in {"so100.buffer.action": ..., "so100-blue.buffer.action": ..., ...}, so the check fails and the tensor is  returned unchanged. Same issue affects state normalization in the preprocessor.

### Reproduction
```
  from lerobot.policies.smolvla.modeling_smolvla import SmolVLAPolicy
  from lerobot.policies.factory import make_pre_post_processors
  import torch

  policy = SmolVLAPolicy.from_pretrained("lerobot/smolvla_base")
  _, postprocessor = make_pre_post_processors(
      policy_cfg=policy.config,
      pretrained_path="lerobot/smolvla_base",
  )

  fake_action = torch.tensor([0.5, 0.5, 0.5, 0.5, 0.5, 0.5])
  result = postprocessor(fake_action)
  print(torch.equal(fake_action, result))  # True — unnormalization was skipped
```

### Evidence

The pre-migration model (revision 3326b100334f) had these layers in its state_dict:

  normalize_inputs.so100_buffer_observation_state.mean: [1.56, 118.65, 110.85, ...]
  normalize_inputs.so100_buffer_observation_state.std:  [26.15, 52.65, 49.38, ...]
  unnormalize_outputs.so100_buffer_action.mean: [1.60, 119.94, 109.77, ...]
  unnormalize_outputs.so100_buffer_action.std:  [26.39, 52.41, 49.85, ...]

The current model (post-migration, revision c83c3163b8ca) has these layers removed, with the stats saved in external safetensors files — but the key mismatch makes them inaccessible.

### Impact
The model core expects z-score inputs and produces z-score outputs, but nothing converts to/from z-scores during inference. This  affects any model trained on multiple datasets whose stats were extracted during the Pipeline migration.

### Suggested fix

In _apply_transform (lerobot/processor/normalize_processor.py), fall back to a suffix match when the exact key is not found:
```
  def _resolve_stats_key(self, key: str) -> str | None:
      """Resolve a feature key to a matching stats key.
   
      Handles the case where stats were saved with dataset-prefixed keys
      (e.g., 'so100.buffer.action') but the feature key is unprefixed ('action').
      """
      if key in self._tensor_stats:
          return key
      # Fallback: find stats keys ending with the feature key
      suffix = f".{key}"
      matches = [k for k in self._tensor_stats if k.endswith(suffix) or k.endswith(f"_{key}")]
      if len(matches) == 1:
          return matches[0]
      if len(matches) > 1:
          logging.warning(
              f"Multiple stats keys match '{key}': {matches}. Using first match."
          )
          return matches[0]
      return None

  Then in _apply_transform, replace:
  if norm_mode == NormalizationMode.IDENTITY or key not in self._tensor_stats:
      return tensor
  with:
  resolved_key = self._resolve_stats_key(key)
  if norm_mode == NormalizationMode.IDENTITY or resolved_key is None:
      return tensor
```

And use resolved_key instead of key for the rest of the method.

**Comment by ceyxasm** [comment]
Additional finding: state normalization stats were dropped during migration 
The pre-migration model had 3 groups of normalization layers:

<img width="1390" height="246" alt="Image" src="https://github.com/user-attachments/assets/7e660fb3-18f5-4d4b-b5dc-5e9f10002c9d" />

The migration only extracted the action stats (12 tensors from normalize_targets + unnormalize_outputs). The 6 normalize_inputs.*_observation_state tensors were not saved to either the preprocessor or postprocessor safetensors files.
```
  # Post-migration preprocessor safetensors — only action keys, no state keys
  so100.buffer.action.mean
  so100.buffer.action.std
  # ... (6 total — all action, zero state)
```
  This means even if the key mismatch (bug 1) is fixed, state normalization would still be broken because the stats simply aren't there.

  The missing state stats from the pre-migration model (revision 3326b100334f):
  normalize_inputs.so100_buffer_observation_state.mean: [1.56, 118.65, 110.85, 56.63, -27.28, 13.95]
  normalize_inputs.so100_buffer_observation_state.std:  [26.15, 52.65, 49.38, 36.57, 58.90, 17.99]

**Comment by zzzxtjjj** [comment]
Hi! I'd like to work on this issue.  
I'll first reproduce the normalization/unnormalization behavior on the current main branch, then investigate the stats-key resolution logic and add a regression test before proposing a fix.  
I'll keep the changes scoped to this issue

**Comment by erikinfo** [comment]
I can confirm that currently the invocation of SmolVLA through LeRobot does not use normalization.
